requester/requester.go

```diff
@@ -111,27 +111,32 @@ func (b *Work) makeRequest(c *http.Client) {
 	var dnsStart, connStart, resStart, reqStart, delayStart time.Time
 	var dnsDuration, connDuration, resDuration, reqDuration, delayDuration time.Duration
 	req := cloneRequest(b.Request, b.RequestBody)
+	redirects := 0
 	if b.EnableTrace {
 		trace := &httptrace.ClientTrace{
 			DNSStart: func(info httptrace.DNSStartInfo) {
 				dnsStart = time.Now()
 			},
 			DNSDone: func(dnsInfo httptrace.DNSDoneInfo) {
-				dnsDuration = time.Now().Sub(dnsStart)
+				dnsDuration += time.Now().Sub(dnsStart)
 			},
 			GetConn: func(h string) {
+				redirects++
+				if redirects > 1 {
```

Comment:
I think if we skip the dns completely as you mentioned in #12 then this would be an overkill as we would have already got the final endpoints link. Else round trip implementation you mentioned above sounds good